adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+        std::cout << "Wrong value " << c[index] << " on element " << i << " "
+                  << j << std::endl;
+        exit(-1);
+      }
+    }
+  }
+
+  std::cout << "Good computation!" << std::endl;
+  return 0;
+}
+----
+
+== New free functions
+
+=== Command-groups
+
+.[apititle]#submit#
+[source,role=synopsis,id=api:submit]
+----
+namespace sycl::khr {
+
+template <typename CommandGroupFunc>
+void submit(sycl::queue q, CommandGroupFunc&& cgf);
+
+}
+----
+_Effects_: Equivalent to [code]#q.submit(cgf)#.
+
+'''
+
+.[apititle]#submit_tracked#
+[source,role=synopsis,id=api:submit_tracked]
+----
+namespace sycl::khr {
+
+template <typename CommandGroupFunc>
+sycl::event submit_tracked(sycl::queue q, CommandGroupFunc&& cgf);
+
+}
+----
+_Effects_: Equivalent to [code]#q.submit(cgf)#.
+
+_Returns_: A [code]#sycl::event# associated with the submitted command.
+
+'''
+
+=== Kernel launch
+
+.[apititle]#launch# (kernel function)
+[source,role=synopsis,id=api:launch]
+----
+namespace sycl::khr {
+
+template <typename KernelType>
+void launch(sycl::handler& h, sycl::range<1> r, const KernelType& k); (1)
+
+template <typename KernelType>
+void launch(sycl::handler& h, sycl::range<2> r, const KernelType& k); (2)
+
+template <typename KernelType>
+void launch(sycl::handler& h, sycl::range<3> r, const KernelType& k); (3)
+
+template <typename KernelType>
+void launch(sycl::queue q, sycl::range<1> r, const KernelType& k);    (4)
+
+template <typename KernelType>
+void launch(sycl::queue q, sycl::range<2> r, const KernelType& k);    (5)
+
+template <typename KernelType>
+void launch(sycl::queue q, sycl::range<3> r, const KernelType& k);    (6)
+
+}
+----
+_Effects (1-3)_: Equivalent to [code]#h.parallel_for(r, k)#.
+
+_Effects (4-6)_: Equivalent to [code]#q.submit([&](handler& h) { launch(h, r,
+k); })#.
+
+'''
+
+.[apititle]#launch# (kernel object)
+[source,role=synopsis,id=api:launch-kernel]
+----
+namespace sycl::khr {
+
+template <typename... Args>
+void launch(sycl::handler& h, sycl::range<1> r,
+            const sycl::kernel& k, Args&&... args); (1)
+
+template <typename... Args>
+void launch(sycl::handler& h, sycl::range<2> r,
+            const sycl::kernel& k, Args&&... args); (2)
+
+template <typename... Args>
+void launch(sycl::handler& h, sycl::range<3> r,
+            const sycl::kernel& k, Args&&... args); (3)
+
+template <typename... Args>
+void launch(sycl::queue q, sycl::range<1> r,
+            const sycl::kernel& k, Args&&... args); (4)
+
+template <typename... Args>
+void launch(sycl::queue q, sycl::range<2> r,
+            const sycl::kernel& k, Args&&... args); (5)
+
+template <typename... Args>
+void launch(sycl::queue q, sycl::range<3> r,
+            const sycl::kernel& k, Args&&... args); (6)
+
+}
+----
+_Effects_: Enqueues a kernel object that is invoked for every work-item in the
+specified [code]#sycl::range#.
+The arguments in [code]#args# are passed to the kernel in the same order.
```

Comment:
The core spec lists a bunch of other semantics when launching a `kernel` object.  Rather than duplicating all the text here, can we say something like:

> _Effects (1) - (3):_ Equivalent to:
> ```
> h.set_args(args...);
> h.parallel_for(r, k);
> ```
> _Effects (4) - (6):_ Equivalent to `q.submit([&](handler& h) { launch(h, r, k, args...); })`

Same for the other APIs below that take a `kernel` object.

Reply:
I've tried to fix this in 437754958e8597148fe1479d7a73691197aa947e.  I'd appreciate it if you could double-check that I got it right.